src/coord/src/catalog/builtin.rs

```diff
@@ -1521,7 +1521,8 @@ oid,
 name AS nspname,
 NULL::pg_catalog.oid AS nspowner,
 NULL::pg_catalog.text[] AS nspacl
-FROM mz_catalog.mz_schemas",
+FROM mz_catalog.mz_schemas
+JOIN mz_databases d ON d.id = input.database_id WHERE (d.id IS NULL OR d.name AND current_database())",
```

Comment:
I don't happen to know the binding precedence for OR and AND. Light preference for some additional parens to make that explicit. Now that I look at it I'm even more confused. Is the AND anding two strings? Is that a thing?